pyaedt/modules/Material.py

```diff
@@ -1992,16 +1993,251 @@ def set_magnetic_coercivity(self, value=0, x=1, y=0, z=0):
+            props["CoefficientSetupData"]["property_data"] = "coreloss_data"
+            props["CoefficientSetupData"]["coefficient_setup"] = coefficient_setup
+            frequency = list(points_list_at_freq.keys())[0]
+            props["CoefficientSetupData"]["Frequency"] = "{}Hz".format(frequency)
+            props["CoefficientSetupData"]["Thickness"] = thickness
+            props["CoefficientSetupData"]["Conductivity"] = str(conductivity)
+            points = list(itertools.chain(*[v for v in list(points_list_at_freq.values())[0]]))
+            props["CoefficientSetupData"]["Coordinates"] = OrderedDict({"DimUnits": ["", ""], "Points": points})
+        elif len(points_list_at_freq.keys()) > 1:
+            props["CoreLossMultiCurveData"] = OrderedDict({})
+            props["CoreLossMultiCurveData"]["property_data"] = "coreloss_multi_curve_data"
+            props["CoreLossMultiCurveData"]["coreloss_unit"] = "w_per_cubic_meter"
+
+            props["CoreLossMultiCurveData"]["AllCurves"] = OrderedDict({})
+            props["CoreLossMultiCurveData"]["AllCurves"]["OneCurve"] = []
+            for freq in list(points_list_at_freq.keys()):
+                points = list(itertools.chain(*[v for v in points_list_at_freq[freq]]))
+                one_curve = OrderedDict(
+                    {
+                        "Frequency": "{}Hz".format(freq),
+                        "Coordinates": OrderedDict({"DimUnits": ["", ""], "Points": points}),
+                    }
+                )
+                props["CoreLossMultiCurveData"]["AllCurves"]["OneCurve"].append(one_curve)
+
+        props = args = self._get_args(props)
+        props.pop(0)
+        coefficients = self.odefinition_manager.ComputeCoreLossCoefficients(
+            core_loss_model_type, self.mass_density.evaluated_value, props[0]
+        )
+        return coefficients
+
+    @pyaedt_function_handler()
+    def set_electrical_steel_coreloss_at_frequency(
+        self,
+        kdc=0,
+        cut_depth="1mm",
+        thickness="0.5mm",
+        conductivity=0,
+        coefficient_setup="w_per_cubic_meter",
+        core_loss_model_type="Electrical Steel",
+        points_list_at_freq={},
+    ):
+        """Set electrical steel core loss model at one single frequency or at multiple frequencies.
 
         Parameters
         ----------
-        kh : float
-        kc : float
-        ke : float
         kdc : float
-        cut_depth : float
+            Coefficient considering the DC flux bias effects
+        cut_depth : str
+            Equivalent cut depth.
+            You use this parameter to consider the manufacturing effects on core loss computation.
+            The default value is ``"1mm"``.
+        thickness : str
+            Thickness specified in terms of the value plus the unit.
+            The default is ``"0.5mm"``.
+        conductivity : float
+            Conductivity. The unit is S/m.
+            The default is ``"0S/m"``.
```

Comment:
Not sure here. Up to you.
```suggestion
            The default is ``"0 S/m"``.
```